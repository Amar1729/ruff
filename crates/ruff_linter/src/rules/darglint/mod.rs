//! Rules from [darglint](https://github.com/terrencepreilly/darglint).
pub(crate) mod rules;
// pub mod settings;

#[cfg(test)]
mod tests {
    use std::path::Path;

    use anyhow::Result;
    use test_case::test_case;

    use crate::assert_messages;
    use crate::registry::Rule;
    use crate::settings::LinterSettings;
    use crate::test::test_path;

    #[test_case(Rule::ExcessParam, Path::new("DAR102.py"))]
    fn rules(rule_code: Rule, path: &Path) -> Result<()> {
        let snapshot = format!("{}_{}", rule_code.noqa_code(), path.to_string_lossy());
        let diagnostics = test_path(
            Path::new("darglint").join(path).as_path(),
            &LinterSettings::for_rule(rule_code),
        )?;
        assert_messages!(snapshot, diagnostics);
        Ok(())
    }
}
