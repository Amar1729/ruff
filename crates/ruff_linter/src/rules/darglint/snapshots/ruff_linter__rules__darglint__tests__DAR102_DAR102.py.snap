---
source: crates/ruff_linter/src/rules/darglint/mod.rs
---
DAR102.py:7:5: DAR102 Missing docstring parameter in the function `foo`: `b`
  |
7 | def foo(a):
  |     ^^^ DAR102
8 |     """
9 |     Args:
  |

DAR102.py:16:5: DAR102 Missing docstring parameters in the function `bar`: `y`, `_z`
   |
16 | def bar(x):
   |     ^^^ DAR102
17 |     """
18 |     Args:
   |

DAR102.py:26:9: DAR102 Missing docstring parameter in the function `method`: `b`
   |
25 | class A:
26 |     def method(self, a):
   |         ^^^^^^ DAR102
27 |         """
28 |         Args:
   |


